.github/workflows/tests.yml

```
@@ -43,3 +43,44 @@ jobs:
                       **/coverage
                       !**/coverage/lcov-report
                       !**/node_modules/**
+
+    playwright:
+        name: Run Playwright end-to-end tests & upload html report
+        runs-on: ubuntu-24.04-arm
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  cache: "yarn"
+                  node-version: "lts/*"
+
+            - name: Install Deps
+              run: "yarn install --frozen-lockfile"
+
+            - name: Get installed Playwright version
+              id: playwright
+              run: echo "version=$(yarn list --pattern @playwright/test --depth=0 --json --non-interactive --no-progress | jq -r '.data.trees[].name')" >> $GITHUB_OUTPUT
+
+            - name: Cache playwright binaries
+              uses: actions/cache@v4
+              id: playwright-cache
+              with:
+                  path: |
+                      ~/.cache/ms-playwright
```

Can this be a single line?
```suggestion
                  path: ~/.cache/ms-playwright
```